core/src/main/java/uk/gov/justice/services/core/annotation/FrameworkInterceptor.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.services.core.annotation;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import uk.gov.justice.services.core.interceptor.Interceptor;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ * Identifies interceptor classes, which are used to intercept incoming JsonEnvelopes before the
+ * dispatcher, or out going messages before they are returned to the adapter.  The class must
+ * implement {@link Interceptor}
+ */
+@Retention(RUNTIME)
+@Target(TYPE)
+@ApplicationScoped
+public @interface FrameworkInterceptor {
```

Comment:
If all classes that have this annotation also have to implement the Interceptor interface then isn't there an easier way of just getting all beans that implement that and using that rather than the annotation injection?

You should be able to intercept the after deployment validation event and find all beans of Type Interceptor ?